lib/src/acter_avatar.dart

```diff
@@ -162,245 +126,242 @@ class _ActerAvatar extends State<ActerAvatar> {
+                      badgesSize),
+              ],
             ),
-            renderSpaceParent(context),
-          ],
-        );
-      case DisplayMode.GroupChat:
-        return Container(
-          height: widget.size ?? 48,
-          width: widget.size ?? 48,
-          decoration: BoxDecoration(
-            borderRadius: BorderRadius.circular(6.0),
-            image: DecorationImage(
-              fit: BoxFit.cover,
-              image: avatar,
-              onError: avatarError,
+        // render for DM
+        DM: (avatarInfo, size) => CircleAvatar(
+              foregroundImage: avatar,
+              onForegroundImageError: avatarError,
+              radius: size ?? 24,
             ),
-          ),
-        );
-      case DisplayMode.GroupDM:
-        return widget.avatarsInfo != null && widget.avatarsInfo!.isNotEmpty
-            ? Stack(
-                alignment: Alignment.bottomLeft,
-                clipBehavior: Clip.none,
-                children: [
-                  CircleAvatar(
-                    foregroundImage: avatar,
-                    onForegroundImageError: avatarError,
-                    radius: widget.size ?? 24,
-                  ),
-                  Positioned(
-                    left: -7,
-                    bottom: -5,
-                    child: widget.secondaryToolTip != TooltipStyle.None
-                        ? Tooltip(
-                            message: secTooltipMsg(),
-                            child: CircleAvatar(
-                              foregroundImage: widget.avatarsInfo![0].avatar,
-                              radius: widget.size ?? 24,
-                            ),
-                          )
-                        : CircleAvatar(
-                            foregroundImage: widget.avatarsInfo![0].avatar,
-                            radius: widget.size ?? 24,
-                          ),
-                  ),
-                  widget.avatarsInfo!.length > 1
-                      ? Positioned.fill(
-                          bottom: -5,
-                          child: Align(
-                            alignment: Alignment.bottomRight,
-                            child: Container(
-                              width: 15,
-                              height: 15,
-                              alignment: Alignment.center,
-                              decoration: const BoxDecoration(
-                                color: Colors.green,
-                                shape: BoxShape.circle,
-                              ),
-                              child: Text(
-                                '+${widget.avatarsInfo!.length - 1}',
-                                style: const TextStyle(fontSize: 8),
+        // render for Group DM
+        GroupDM: ((avatarInfo, groupAvatarsInfo, size, groupAvatarSize) {
+          return groupAvatarsInfo != null && groupAvatarsInfo.isNotEmpty
```

Comment:
I wonder if we shouldn't move all these into separate functions to make it easier to keep the overview of what is happenening when ... so for every when have a corresponding function being called with the parameters given ...

Reply:
I think we could move these into sub widget methods to enhance code readability.